packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -33,15 +37,43 @@ export const createEntity: ResolverFn<
    * @see https://app.asana.com/0/1202805690238892/1203084714149803/f
    */
 
-  const entity = await EntityModel.createEntityWithLinks(graphApi, {
-    ownedById: ownedById ?? userModel.getEntityUuid(),
-    entityTypeId,
-    properties,
-    linkedEntities: linkedEntities ?? undefined,
-    actorId: userModel.getEntityUuid(),
-  });
+  let entityModel: EntityModel | LinkEntityModel;
+
+  if (linkMetadata) {
+    const { leftEntityId, leftOrder, rightEntityId, rightOrder } = linkMetadata;
+
+    const [leftEntityModel, rightEntityModel, linkEntityTypeModel] =
+      await Promise.all([
+        EntityModel.getLatest(graphApi, {
+          entityId: leftEntityId,
+        }),
+        EntityModel.getLatest(graphApi, {
+          entityId: rightEntityId,
+        }),
+        EntityTypeModel.get(graphApi, { entityTypeId }),
+      ]);
+
+    entityModel = await LinkEntityModel.createLinkEntity(graphApi, {
+      leftEntityModel,
+      leftOrder: leftOrder ?? undefined,
+      rightEntityModel,
+      rightOrder: rightOrder ?? undefined,
+      properties,
+      linkEntityTypeModel,
+      ownedById: ownedById ?? userModel.getEntityUuid(),
+      actorId: userModel.getEntityUuid(),
+    });
```

Comment:
Just something to think about when reworking these model classes soon. I definitely don't think we need to be querying the endpoint entities just to create the thing, we're not even using this information from what I can see, just passing it in to immediately drop it again

Reply:
Agreed